CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/jdbc/ImportJDBCPlugin.java

```diff
@@ -0,0 +1,305 @@
+                }
+
+                // Determine if a positional attribute has been defined, if so update the overall flag
+                final boolean isPositional = attributeDefintionIsPositional(definition.getDefinitions(AttributeType.SOURCE_VERTEX), definition.getDefinitions(AttributeType.DESTINATION_VERTEX));
+                positionalAtrributesExist = (positionalAtrributesExist || isPositional);
+            }
+
+            // If at least one positional attribute has been received for either the src or destination vertex we will assume that the user is trying to import positions and won't auto arrange
+            // the graph. This does mean some nodes could sit on top of each other if multiple nodes have the same coordinates.
+            if (!positionalAtrributesExist) {
+                interaction.setProgress(1, 1, "Arranging", true);
+
+                graph.validateKey(GraphElementType.VERTEX, true);
+                graph.validateKey(GraphElementType.TRANSACTION, true);
+
+                // unfortunately need to arrange with pendants and uncollide because grid arranger works based on selection
+                final VertexListInclusionGraph vlGraph = new VertexListInclusionGraph(graph, AbstractInclusionGraph.Connections.NONE, newVertices);
+                PluginExecutor.startWith(ArrangementPluginRegistry.GRID_COMPOSITE)
+                        .followedBy(ArrangementPluginRegistry.PENDANTS)
+                        .followedBy(ArrangementPluginRegistry.UNCOLLIDE)
+                        .executeNow(vlGraph.getInclusionGraph());
+                vlGraph.retrieveCoords();
+            }
+        }
+    }
+
+    // If src or destination attribute definitions have been supplied, check them and return true if any of the positional attributes ('x', 'y', or 'z') are included. Positional
+    // arguments allow the import to define where nodes will be placed on graph. If src or destination definitons do not exist then an empty list should be supplied.
+    private static boolean attributeDefintionIsPositional(final List<ImportAttributeDefinition> srcAttributeDefinitions, final List<ImportAttributeDefinition> destAttributeDefinitions) {
+
+        // Check if srcAttributeDefintions contain positional attributes
+        if (srcAttributeDefinitions.stream().map(attribute -> attribute.getAttribute().getName()).anyMatch(name -> (VisualConcept.VertexAttribute.X.getName().equals(name) || VisualConcept.VertexAttribute.Y.getName().equals(name) || VisualConcept.VertexAttribute.Z.getName().equals(name)))) {
+            return true;
+        }
+        // Check if destAttributeDefintions contain positional attributes
+        return destAttributeDefinitions.stream().map(attribute -> attribute.getAttribute().getName()).anyMatch(name -> (VisualConcept.VertexAttribute.X.getName().equals(name) || VisualConcept.VertexAttribute.Y.getName().equals(name) || VisualConcept.VertexAttribute.Z.getName().equals(name)));
```

Comment:
This could be simplified by or'ing the 2 boolean expressions:
`return srcAttributeDefinitions... || destAttributeDefinitions...`

Reply:
I changed this to or over the expressions and to me it seems less readable. Can still go ahead with the change

Reply:
You can always put the 2nd expression on a new line. Personally I prefer my suggestion as it removes the need for the if block

Reply:
Done, I've also added to the whatsnew.txt and fixed that last naming convention issue.